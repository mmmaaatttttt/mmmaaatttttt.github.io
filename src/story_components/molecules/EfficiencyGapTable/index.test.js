import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { EfficiencyGapTable } from ".";

describe("smoke and snapshot tests", () => {
  it("renders successfully", () => {
    shallow(<EfficiencyGapTable />);
  });

  it("renders with no district counts", () => {
    const wrapper = shallow(<EfficiencyGapTable />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders with district counts", () => {
    const wrapper = shallow(
      <EfficiencyGapTable
        districtCounts={[[[5, 4], [5, 4], [5, 4], [5, 4], [2, 7], [5, 4]]]}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
