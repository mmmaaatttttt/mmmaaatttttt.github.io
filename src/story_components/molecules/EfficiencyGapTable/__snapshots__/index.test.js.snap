// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests renders with district counts 1`] = `
<div>
  <p>
    Here's a sample efficiency gap calculation based on the districts you created above.
  </p>
  <Styled(styled.table)
    margin="0 0 1.44rem 0"
  >
    <thead>
      <tr>
        <th>
          District
        </th>
        <th>
          Wasted Votes
        </th>
        <th>
          Wasted Votes
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        key="0"
      >
        <td>
          1
        </td>
        <td>
          5
          4
        </td>
        <td>
          NaN
        </td>
      </tr>
      <tr>
        <td>
          Total
        </td>
        <td>
          05,4
        </td>
        <td>
          NaN
        </td>
      </tr>
      <tr>
        <td>
          Efficiency Gap
        </td>
        <td
          colSpan="2"
        >
          <div>
            (
            <styled.span
              bold={false}
              color="#2227FF"
            >
              05,4
            </styled.span>
             
            –
             
            <styled.span
              bold={false}
              color="#FF3C23"
            >
              NaN
            </styled.span>
            ) ÷ 
            05,45,4
             = 
            NaN
            %
             in favor of red
            .
          </div>
        </td>
      </tr>
    </tbody>
  </Styled(styled.table)>
</div>
`;

exports[`smoke and snapshot tests renders with no district counts 1`] = `
<p>
  <b>
    To see a sample calculation of the efficiency gap, please finish drawing your districts above.
  </b>
</p>
`;
